### ISSUE TITLE
[Bug]: Tablist with Overflow breaks when orientation is changed from vertical to horizontal & vice versa

### ISSUE THREAD

**Issue by Ramakrishnan24689** [microsoft/fluentui#27046]
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
"@fluentui/react-components": "9.15.6"
```

### Are you reporting Accessibility issue?

None

### Reproduction

https://codesandbox.io/s/priceless-surf-viocof?file=/example.tsx

## Actual Behavior
Tablist with Overflow breaks when orientation is changed from vertical to horizontal & vice versa.
In the codesandbox sample, change the orientation to vertical by enabling the switch and select Agenda tab.
Control breaks with an error "Cannot read properties of undefined (reading 'element')"

## Expected Behavior
Tablist with Overflow should not break even if orientation is changed dynamically.

### Logs

```shell
The issue found in OverflowManager.js file where at one point itemB is undefined. 

 const visibleItemQueue = priorityQueue_1.createPriorityQueue((a, b) => {
    const itemA = overflowItems[a];
    const itemB = overflowItems[b];
 // Lower priority at the top of the queue
    const priority = itemA.priority - itemB.priority;
    if (priority !== 0) {
      return priority;
    }
```

### Requested priority

Blocking

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by GeoffCoxMSFT** [comment]
@ling1726  - I'll investigate.  Assigning you too in case you have some insight as the overflow expert.

**Comment by GeoffCoxMSFT** [comment]
@ling1726 - Playing around with just the Overflow Vertical story, the layout doesn't update correctly when toggling vertical and horiztonal on the axis.  I tried exposing forceUpdate on context and calling that when changing the axis, but it didn't help.  Is the orientationAxis built to be updated at runtime?  Suggestions for where a useEffect or size cache clearing might go when this prop changes?

**Comment by ling1726** [comment]
> @ling1726 - Playing around with just the Overflow Vertical story, the layout doesn't update correctly when toggling vertical and horiztonal on the axis. I tried exposing forceUpdate on context and calling that when changing the axis, but it didn't help. Is the orientationAxis built to be updated at runtime? Suggestions for where a useEffect or size cache clearing might go when this prop changes?

Thanks for doing the initial dive in and the helpful insights. I took a deeper look and yeah, this scenario is definitely broken. There are things that should have been disposed but weren't.

I made a start to a fix in #29375
